js_modules/dagit/packages/core/src/launchpad/LaunchRootExecutionButton.tsx

```diff
@@ -47,7 +54,14 @@ export const LaunchRootExecutionButton: React.FC<LaunchRootExecutionButtonProps>
     try {
       const result = await launchPipelineExecution({variables});
       logTelemetry(TelemetryAction.LAUNCH_RUN, metadata);
-      handleLaunchResult(basePath, props.pipelineName, result, {});
+      if (
+        result.data?.launchPipelineExecution.__typename === 'LaunchRunSuccess' &&
+        !props.showRunAfterLaunch
+      ) {
+        showLaunchSuccessToast(history, result.data.launchPipelineExecution);
+      } else {
+        handleLaunchResult(basePath, props.pipelineName, result, {});
```

Comment:
The name `handleLaunchResult` and the fact that its already doing a bunch of if else nav stuff based on the response type made me wonder if we should just fold the code above into handleLaunchresult and make showRunAfter launch an opt (the last param in handleLaunchResult)

Reply:
That's a really good point actually... It looks like that handleLaunchResult method already has an "open in new tab" option, so this should really just be a "toast only" option to that same method. Going to see if I can rework this a bit!